File: Q21 Majority Element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n = nums.length;
        int key = nums[0], count = 0;
        for (int i = 0; i < n; i++) {
            if (count == 0) {
                key = nums[i];
            }
            if (nums[i] == key) {
                count++;
            } else {
                count--;
            }
        }
        return key;
    }
}

File: Q21 Majority Element.py
class Solution:
    def majorityElement(self, nums):
        n = len(nums)
        key = nums[0]
        count = 0
        for i in range(n):
            if count == 0:
                key = nums[i]
            if nums[i] == key:
                count += 1
            else:
                count -= 1
        return key
